### ISSUE TITLE
Wait for page load before saving in case translator becomes available

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero-connectors#105]
[Moved from https://github.com/zotero/zotero/issues/868 now that we're connector-only]

Since detection doesn't run until the page loads, it's pretty easy to save a webpage item when better metadata would actually be available shortly. It might make sense to delay saving if the page is loading and then automatically use the default translator if one is available.

On some pages, loading might never finish, or might take a long time, so there should be a time limit to keep the save from hanging forever.

The progress window might say "Waiting for page load…" or something to convey what it's doing (and maybe hint to some people that they can stop it manually).

**Comment by gagarine** [comment]
Agree it's a problem. I tend to refresh the page when Zotero didn't detect the content type (append a lot on Amazon).

What I don't understand is why translator need a page fully loaded. I feel like the page's HTML is enough no? Do we really need a dom rendering?

**Comment by dstillman** [comment]
> Do we really need a dom rendering?

Yes, or translators wouldn't work on any site that generated content via JS.

(In the next connector version the DOM will no longer be available when saving multiple results, so some translators may be adjusted to make necessary XHR requests/etc. themselves for a single save too in order to share the same code, which would make this less necessary, but at least for the foreseeable future this will remain necessary.)
